server/get_payload.go

```diff
@@ -36,16 +38,39 @@ var (
 	errInvalidBlockhash = errors.New("invalid blockhash")
 	errInvalidKZGLength = errors.New("invalid KZG commitments length")
 	errInvalidKZG       = errors.New("invalid KZG commitment")
-	errFailedToDecode   = errors.New("failed to decode payload")
 	errFailedToConvert  = errors.New("failed to convert block from SSZ to JSON")
 )
 
+type GetPayloadVersion string
+
+const (
+	GetPayloadV1 GetPayloadVersion = "V1"
+	GetPayloadV2 GetPayloadVersion = "V2"
+)
+
 ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 // Core Logic
 ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
 
+type payloadResult struct {
+	success  bool
+	response *builderApi.VersionedSubmitBlindedBlockResponse
+}
+
+// Deprecated: For reference: https://github.com/ethereum/builder-specs/issues/119
 // getPayload requests the payload (execution payload, blobs bundle, etc) from the relays
 func (m *BoostService) getPayload(log *logrus.Entry, signedBlindedBeaconBlockBytes []byte, userAgent, proposerContentType, proposerAcceptContentTypes, proposerEthConsensusVersion string) (*builderApi.VersionedSubmitBlindedBlockResponse, bidResp) {
+	result, bid := m.innerGetPayload(log, signedBlindedBeaconBlockBytes, userAgent, proposerContentType, proposerAcceptContentTypes, proposerEthConsensusVersion, GetPayloadV1)
+	return result.response, bid
+}
+
+// getPayloadV2 submits the signed blinded beacon block to relays for submission without returning the payload and blobs
```

Comment:
is it necessary to have two functions here? feels like it could be a single function that returns all fields, since the methods themselves are identical